app/services/re_captcha/check_enabled.rb

```diff
@@ -0,0 +1,37 @@
+# This service encapsulates the logic related to validating if reCAPTCHA is
+# enabled in the current Forem instance. The decision is based on making
+# sure the necessary SiteConfig keys are available and also on the user
+# object passed in.
+#
+# Example use: ReCaptcha::CheckEnabled.call(current_user) => true/false
+module ReCaptcha
+  class CheckEnabled
+    def self.call(current_user = nil)
+      new(current_user).call
+    end
+
+    def initialize(current_user)
+      @current_user = current_user
+    end
+
+    def call
+      # recaptcha will not be enabled if site key and secret key aren't set
+      return false unless keys_configured?
+      # recaptcha will always be enabled when not logged in
+      return true if @current_user.nil?
```

Comment:
can we invert these? This way we don't risk hitting the DB with `keys_configured?` for anonymous users if we already know it's always enabled for them

Reply:
Interesting idea but I don't think we can, because if the user is `nil` but the reCAPTCHA keys are not configured we can't show it (it will be broken). The user won't be able to pass the reCAPTCHA and it will render as an error in the form, somewhat like this:

![Screen Shot 2020-12-02 at 11 09 32](https://user-images.githubusercontent.com/6045239/100906557-e95b3080-348e-11eb-8427-e94948c419f4.png)

In other words, if we want to render the reCAPTCHA prompt we need to check the `SiteConfig` keys are configured.

When rendering the reCAPTCHA (in the view) Rails will hit the DB regardless (in order to pass in the site_key config) so they will be cached (AFAICT `SiteConfig` will do that internally) so at least it won't be hit multiple times for the same keys, right?

Reply:
makes sense! thanks for the explanation :)